kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -359,53 +386,148 @@ protected ProcurementCardDefault retrieveProcurementCardDefault(String creditCar
      */
     @Override
     public boolean autoApproveProcurementCardDocuments() {
-        //if auto approve is turned on
-        boolean autoApproveOn = super.parameterService.getParameterValueAsBoolean(
-                ProcurementCardAutoApproveDocumentsStep.class,
-                ProcurementCardDocumentRuleConstants.AUTO_APPROVE_DOCUMENTS_IND);
+        // if auto approve is turned on
+        boolean autoApproveOn = parameterService.getParameterValueAsBoolean(ProcurementCardAutoApproveDocumentsStep.class, ProcurementCardDocumentRuleConstants.AUTO_APPROVE_DOCUMENTS_IND);
 
-        if (!autoApproveOn) { // no auto approve?  then skip out of here...
+        if (!autoApproveOn) { // no auto approve? then skip out of here...
             return true;
         }
 
+        StringBuffer autoApproveErrorStr = new StringBuffer();
+
+        // get Parameter information
+        Timestamp currentDate = dateTimeService.getCurrentTimestamp();
+        int autoApproveNumberDays = Integer.parseInt(parameterService.getParameterValueAsString(ProcurementCardAutoApproveDocumentsStep.class, ProcurementCardDocumentRuleConstants.AUTO_APPROVE_NUMBER_OF_DAYS));
```

Comment:
Please refactor the one line parameter accessing and conversion to Integer to:
String autoApproveDaysParam = parameterService.getParameterValueAsString(ProcurementCardAutoApproveDocumentsStep.class, ProcurementCardDocumentRuleConstants.AUTO_APPROVE_NUMBER_OF_DAYS);
        int autoApproveNumberDays
         if (StringUtils.isNotEmpty(autoApproveDaysParam) && StringUtils.isNumeric(autoApproveDaysParam)) {
             autoApproveNumberDays = (Integer.parseInt(autoApproveDaysParam));
         }
         else {
            LOG.debug("Invalid value for autoApproveDaysParam: " + autoApproveDaysParam + ". Can't process auto-approve.");
              return false; // can't process auto approve? -> you might want to check with Julie/Misty if we want to stop processing if we can't get a reliable value for this param... or just return true...
         }

_Tech Note:_
I've alredy seen the above reading and transformation of a param from string to int in at least 2 other places in the Pcard job. I think at this point it's worth creating a utility method somewhere that will do this job.